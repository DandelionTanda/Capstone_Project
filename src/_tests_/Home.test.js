import 'react-native'
import React from 'react'
import Home, {Item} from "../screens/Home"
import renderer from 'react-test-renderer';
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import { View, Text, Pressable } from 'react-native';
import { act } from 'react-test-renderer';
import {fakeUser_single, fakeUser_multiple, fakeDiscounts, fakeOnDiscounts, fakeOffDiscounts, fakeClockins_onShift, fakeClockins_offShift} from './fakeData'
jest.mock('@react-navigation/native');
describe('When single-job user clocked in the company having discounts and successfully fetch data', () => {
  
  beforeEach(() => {   
    global.fetch = jest
    .fn()
    .mockImplementationOnce(() => Promise.resolve(
      Promise.resolve({
        ok: true,
        status: 200,       
        json: () => Promise.resolve(fakeUser_single)
      })
    )).mockImplementationOnce(() => Promise.resolve(
      Promise.resolve({
        ok: true,
        status: 200,       
        json: () => Promise.resolve(fakeDiscounts)
      })
    )).mockImplementationOnce(() => Promise.resolve(
      Promise.resolve({
        ok: true,
        status: 200,       
        json: () => Promise.resolve(fakeClockins_onShift)
      })
    ));        
        
  });

  it('should render properly', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.toJSON()).toMatchSnapshot();
      
    })
        
  })

  it('should render discount list', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId('discount-list')).toBeTruthy();
      
    })
        
  })

  it('should display clocked in', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("clocked in")).toBeTruthy();      
      expect(screen.queryByText("clocked out")).toBeFalsy();
      
    })       
  })

  it('should display on-shift discounts', async () => {
    const screen = render(<Home />);
    await waitFor(() => {    
      expect(screen.queryByTestId("On-shift Discount - 1")).toBeTruthy(); 
      expect(screen.queryByTestId("On-shift Discount - 2")).toBeTruthy(); 
      expect(screen.queryByTestId("Off-shift Discount - 1")).toBeFalsy(); 
      expect(screen.queryByTestId("Off-shift Discount - 2")).toBeFalsy();
    })
        
  })
  it('should not display drop menu', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId("switch-company-drop-menu")).toBeFalsy();   
           
    })       
  })  

  it('should not display error and refresh button', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("Error message")).toBeFalsy();   
      expect(screen.queryByText("Refresh")).toBeFalsy();       
    })       
  })  

  it('should navigate to dscount screen after pressing first discount item', async () => {        
    // Mocking navigate method
    const navigate = jest.fn();
    // Getting element
    const screen = render(<Home navigation={{ navigate }} />);
    await waitFor(() => {
      
      const discount = screen.queryByTestId('On-shift Discount - 1');
      fireEvent.press(discount);
    });   
    // Asserting screen navigation
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('Discount', {discount:fakeOnDiscounts[0], user:fakeUser_single});
  });  

  it('should navigate to dscount screen after pressing last discount item', async () => {        
    // Mocking navigate method
    const navigate = jest.fn();
    // Getting element
    const screen = render(<Home navigation={{ navigate }} />);
    await waitFor(() => {
      
      const discount = screen.queryByTestId(`On-shift Discount - ${fakeOnDiscounts.length}`);
      fireEvent.press(discount);
    });   
    // Asserting screen navigation
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('Discount', {discount:fakeOnDiscounts[fakeOnDiscounts.length-1], user:fakeUser_single});
  });  
  
})

describe('When single-job user clocked out the company having discounts and successfully fetch data', () => {
  beforeEach(() => {
      global.fetch = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(
        Promise.resolve({
          ok: true,
          status: 200,       
          json: () => Promise.resolve(fakeUser_single)
        })
      )).mockImplementationOnce(() => Promise.resolve(
        Promise.resolve({
          ok: true,
          status: 200,       
          json: () => Promise.resolve(fakeDiscounts)
        })
      )).mockImplementationOnce(() => Promise.resolve(
        Promise.resolve({
          ok: true,
          status: 200,       
          json: () => Promise.resolve(fakeClockins_offShift)
        })
      ));     
    
       
  })  
  it('should render properly', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.toJSON()).toMatchSnapshot();
      
    })
        
  })

  it('should render discount list', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId('discount-list')).toBeTruthy();
      
    })
        
  })

  it('should display clocked out', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("clocked in")).toBeFalsy();   
      expect(screen.queryByText("clocked out")).toBeTruthy(); 
      
    })       
  })

  it('should display off-shift discounts', async () => {
    const screen = render(<Home />);
    await waitFor(() => {    
      expect(screen.queryByTestId("On-shift Discount - 1")).toBeFalsy();
      expect(screen.queryByTestId("On-shift Discount - 2")).toBeFalsy();
      expect(screen.queryByTestId("Off-shift Discount - 1")).toBeTruthy(); 
      expect(screen.queryByTestId("Off-shift Discount - 2")).toBeTruthy(); 
    })
        
  })

  it('should not display drop menu', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId("switch-company-drop-menu")).toBeFalsy();   
           
    })       
  })  

  it('should not display error and refresh button', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("Error message")).toBeFalsy();   
      expect(screen.queryByText("Refresh")).toBeFalsy();       
    })       
  })

  it('should navigate to dscount screen after pressing first discount item', async () => {   
  
    // Mocking navigate method
    const navigate = jest.fn();
    // Getting element
    const screen = render(<Home navigation={{ navigate }} />);
    await waitFor(() => {
      
      const discount = screen.queryByTestId('Off-shift Discount - 1');
      fireEvent.press(discount);
    });
    // Asserting screen navigation
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('Discount', {discount:fakeOffDiscounts[0], user:fakeUser_single});
  });  

  it('should navigate to dscount screen after pressing last discount item', async () => {        
    // Mocking navigate method
    const navigate = jest.fn();
    // Getting element
    const screen = render(<Home navigation={{ navigate }} />);
    await waitFor(() => {
      
      const discount = screen.queryByTestId(`Off-shift Discount - ${fakeOffDiscounts.length}`);
      fireEvent.press(discount);
    });   
    // Asserting screen navigation
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('Discount', {discount:fakeOffDiscounts[fakeOffDiscounts.length-1], user:fakeUser_single});
  });  

  
  
})

describe('When user clocked out the company having discounts and failed to fetch data', () => {
  beforeEach(() => {
      global.fetch = jest
      .fn()
      .mockImplementationOnce(() => Promise.reject(
        new Error('An error has occured: 400')
      )).mockImplementationOnce(() => Promise.reject(
        new Error('An error has occured: 400')
      )).mockImplementationOnce(() => Promise.reject(
        new Error('An error has occured: 400')
      ));
       
  })  
  it('should render properly', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.toJSON()).toMatchSnapshot();
      
    })
        
  })

  it('should not render discount list', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId('discount-list')).toBeFalsy();
      
    })
        
  })

  it('should not display clocked in/out', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("clocked in")).toBeFalsy();   
      expect(screen.queryByText("clocked out")).toBeFalsy(); 
      
    })       
  })

  it('should not display drop menu', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId("switch-company-drop-menu")).toBeFalsy();   
           
    })       
  })  

  it('should display error and refresh button', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText('An error has occured: 400')).toBeTruthy();   
      expect(screen.queryByText("Refresh")).toBeTruthy();       
    })       
  })
  
})


it('Should make item in discount list be pressable', async () => {
  
    const onPress = jest.fn();
    const { getByText } = render(
      <Item item={item = { name: 'Test' }} onPress={onPress} />
    );     
    expect(getByText('Test')).toBeTruthy();
      //expect(onPress).toHaveBeenCalled();    
        
})

describe('When multiple-job user clocked out the authorized company having discounts and successfully fetch data', () => {
  beforeEach(() => {
      global.fetch = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(
        Promise.resolve({
          ok: true,
          status: 200,       
          json: () => Promise.resolve(fakeUser_multiple)
        })
      )).mockImplementationOnce(() => Promise.resolve(
        Promise.resolve({
          ok: true,
          status: 200,       
          json: () => Promise.resolve(fakeDiscounts)
        })
      )).mockImplementationOnce(() => Promise.resolve(
        Promise.resolve({
          ok: true,
          status: 200,       
          json: () => Promise.resolve(fakeClockins_offShift)
        })
      ));          
  })  

  it('should render properly', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.toJSON()).toMatchSnapshot();
      
    })
        
  })

  it('should render discount list', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId('discount-list')).toBeTruthy();
      
    })
        
  })

  it('should display clocked out', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("clocked in")).toBeFalsy();   
      expect(screen.queryByText("clocked out")).toBeTruthy(); 
      
    })       
  })

  it('should display off-shift discounts', async () => {
    const screen = render(<Home />);
    await waitFor(() => {    
      expect(screen.queryByTestId("On-shift Discount - 1")).toBeFalsy();
      expect(screen.queryByTestId("On-shift Discount - 2")).toBeFalsy();
      expect(screen.queryByTestId("Off-shift Discount - 1")).toBeTruthy(); 
      expect(screen.queryByTestId("Off-shift Discount - 2")).toBeTruthy(); 
    })
        
  })

  it('should display drop menu', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByTestId("switch-company-drop-menu")).toBeTruthy();   
           
    })       
  })  

  it('should store organisations in drop menu', async () => {
   
    const screen = render(<Home />);
    await waitFor(() => {    
      let dropdown = screen.queryByTestId("switch-company-drop-menu");
      let items = dropdown.props.items;
      expect(items[0].value).toBe(fakeUser_multiple.organisations[0].id);   
      expect(items[0].label).toBe(fakeUser_multiple.organisations[0].name);

      expect(items[1].value).toBe(fakeUser_multiple.organisations[1].id);        
      expect(items[1].label).toBe(fakeUser_multiple.organisations[1].name);  

      expect(items[2].value).toBe(fakeUser_multiple.organisations[2].id);      
      expect(items[2].label).toBe(fakeUser_multiple.organisations[2].name);  
      
    })       
  })  

  it('should not display error and refresh button', async () => {
    const screen = render(<Home />);
    await waitFor(() => {     
      expect(screen.queryByText("Error message")).toBeFalsy();   
      expect(screen.queryByText("Refresh")).toBeFalsy();       
    })       
  })

  it('should navigate to dscount screen after pressing first discount item', async () => {   
  
    // Mocking navigate method
    const navigate = jest.fn();
    // Getting element
    const screen = render(<Home navigation={{ navigate }} />);
    await waitFor(() => {
      
      const discount = screen.queryByTestId('Off-shift Discount - 1');
      fireEvent.press(discount);
    });
    // Asserting screen navigation
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('Discount', {discount:fakeOffDiscounts[0], user:fakeUser_multiple});
  });  

  it('should navigate to dscount screen after pressing last discount item', async () => {        
    // Mocking navigate method
    const navigate = jest.fn();
    // Getting element
    const screen = render(<Home navigation={{ navigate }} />);
    await waitFor(() => {
      
      const discount = screen.queryByTestId(`Off-shift Discount - ${fakeOffDiscounts.length}`);
      fireEvent.press(discount);
    });   
    // Asserting screen navigation
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('Discount', {discount:fakeOffDiscounts[fakeOffDiscounts.length-1], user:fakeUser_multiple});
  });  

  
  
})
