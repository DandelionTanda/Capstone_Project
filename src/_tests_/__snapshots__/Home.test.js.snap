// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When user clocked in the company having discounts and successfully fetch data should render properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F3F3",
      "flex": 1,
      "marginBottom": 24,
      "marginLeft": 24,
      "marginRight": 24,
      "marginTop": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "green",
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 15,
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "color": "white",
          "height": 52,
          "padding": 10,
          "paddingLeft": 15,
          "width": "37%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "fontWeight": "bold",
          }
        }
        title="clocked in"
      >
        clocked in
      </Text>
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={
      Array [
        Object {
          "id": 124987341,
          "model_id": 1104624,
          "name": "On-shift Discount - 1",
          "object_name": "Discount",
          "onshift": true,
          "updated_at": 1629077363,
          "uuid": "1dc0fa90-8076-43a7-b7cb-c537466f0d72",
          "value": "25%",
        },
        Object {
          "id": 124978219,
          "model_id": 1104624,
          "name": "On-shift Discount - 2",
          "object_name": "Discount",
          "onshift": true,
          "updated_at": 1629075239,
          "uuid": "e443043b-91e6-49b4-a269-2d9e2f9c8598",
          "value": "35%",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginVertical": 10,
      }
    }
    testID="discount-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#45B8DB",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginVertical": 16,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="On-shift Discount - 1"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "width": "70%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            On-shift Discount - 1
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "height": "100%",
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "textAlign": "center",
                  "width": "25%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            25%
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#45B8DB",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginVertical": 16,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="On-shift Discount - 2"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "width": "70%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            On-shift Discount - 2
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "height": "100%",
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "textAlign": "center",
                  "width": "25%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            35%
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`When user clocked out the company having discounts and failed to fetch data should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "red",
        "fontSize": 20,
        "justifyContent": "center",
      }
    }
  >
    An error has occured: 400
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#45B8DB",
        "borderRadius": 4,
        "justifyContent": "center",
        "marginLeft": 20,
        "marginRight": 20,
        "opacity": 1,
        "paddingVertical": 16,
        "position": "relative",
        "top": 20,
        "width": "40%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Refresh
    </Text>
  </View>
</View>
`;

exports[`When user clocked out the company having discounts and successfully fetch data should render properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F3F3",
      "flex": 1,
      "marginBottom": 24,
      "marginLeft": 24,
      "marginRight": 24,
      "marginTop": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "red",
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 15,
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "color": "white",
          "height": 52,
          "padding": 10,
          "width": "37%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 19,
            "fontWeight": "bold",
          }
        }
        title="clocked out"
      >
        clocked out
      </Text>
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={
      Array [
        Object {
          "id": 125572990,
          "model_id": 1104624,
          "name": "Off-shift Discount - 1",
          "object_name": "Discount",
          "onshift": false,
          "updated_at": 1629331565,
          "uuid": "dcec8241-5e5f-4a5c-b88c-0245f11407ff",
          "value": "35%",
        },
        Object {
          "id": 124886241,
          "model_id": 1104624,
          "name": "Off-shift Discount - 2",
          "object_name": "Discount",
          "onshift": false,
          "updated_at": 1629025638,
          "uuid": "0c0abdd2-a0c9-4242-8515-b885db8d1efb",
          "value": "40%",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginVertical": 10,
      }
    }
    testID="discount-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#45B8DB",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginVertical": 16,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="Off-shift Discount - 1"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "width": "70%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Off-shift Discount - 1
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "height": "100%",
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "textAlign": "center",
                  "width": "25%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            35%
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#45B8DB",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginVertical": 16,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="Off-shift Discount - 2"
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "width": "70%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Off-shift Discount - 2
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "height": "100%",
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 19,
                  "textAlign": "center",
                  "width": "25%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            40%
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
